packages/asset/contracts/AssetCreate.sol

```diff
@@ -0,0 +1,336 @@
+    EIP712Upgradeable,
+    AccessControlUpgradeable
+{
+    using TokenIdUtils for uint256;
+
+    IAsset private assetContract;
+    ICatalyst private catalystContract;
+    AuthValidator private authValidator;
+
+    // mapping of creator address to creator nonce, a nonce is incremented every time a creator mints a new token
+    mapping(address => uint16) public creatorNonces;
+
+    bytes32 public constant SPECIAL_MINTER_ROLE =
+        keccak256("SPECIAL_MINTER_ROLE");
+    bytes32 public constant BRIDGE_MINTER_ROLE =
+        keccak256("BRIDGE_MINTER_ROLE");
+    bytes32 public constant MINT_TYPEHASH =
+        keccak256(
+            "Mint(address creator,uint16 nonce,uint8 tier,uint256 amount,bool revealed,string metadataHash)"
+        );
+    bytes32 public constant MINT_BATCH_TYPEHASH =
+        keccak256(
+            "Mint(address creator,uint16 nonce,uint8[] tiers,uint256[] amounts,bool[] revealed,string[] metadataHashes)"
+        );
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the contract
+    /// @param _assetContract The address of the asset contract
+    /// @param _authValidator The address of the AuthValidator contract
+    /// @param _forwarder The address of the forwarder contract
+    function initialize(
+        string memory _name,
+        string memory _version,
+        address _assetContract,
+        address _catalystContract,
+        address _authValidator,
+        address _forwarder,
+        address _defaultAdmin
+    ) public initializer {
+        assetContract = IAsset(_assetContract);
+        catalystContract = ICatalyst(_catalystContract);
+        authValidator = AuthValidator(_authValidator);
+        __ERC2771Handler_initialize(_forwarder);
+        __EIP712_init(_name, _version);
+        __AccessControl_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, _defaultAdmin);
+    }
+
+    /// @notice Create a new asset
+    /// @param signature A signature generated by TSB
+    /// @param tier The tier of the asset to mint
+    /// @param amount The amount of the asset to mint
+    /// @param metadataHash The metadata hash of the asset to mint
+    function createAsset(
+        bytes memory signature,
+        uint8 tier,
+        uint256 amount,
+        bool revealed,
+        string calldata metadataHash
+    ) external {
+        address creator = _msgSender();
+        require(
+            authValidator.verify(
+                signature,
+                _hashMint(creator, tier, amount, revealed, metadataHash)
+            ),
+            "Invalid signature"
+        );
+
+        uint16 creatorNonce = getCreatorNonce(creator);
+        uint256 tokenId = TokenIdUtils.generateTokenId(
+            creator,
+            tier,
+            creatorNonce,
+            revealed ? 1 : 0
+        );
+
+        // burn catalyst of a given tier
+        catalystContract.burnFrom(creator, tier, amount);
```

Comment:
Qu: who is expected to use AssetCreate - will this still be AssetMinter? So we need `burnFrom` and not just `burn`?

Reply:
Thanks. So why do we need `burnFrom` and not just `burn` if the user will call the function? Perhaps this is a naming issue -  the concept of burnFrom should be: an approved operator / admin burns token on behalf of a user. The concept of burn should be: user burns own token.

Reply:
While in the AssetCreate the _msgSender() is read correctly even if submitted by a forwarder, catalyst does its own _msgSender() check and it takes the AssetCreate address as the address to burn from since its not directly called by the user.

And that is technically correct behaviour since its  smart contract <> smart contract call.